docs/advanced/compute-recovery.md

```diff
@@ -0,0 +1,42 @@
+# Recovering from machine failures
+
+Status: **General Availability**
+
+In a cloud environment, machines can fail at any time. Differential transparently handles machine failures, and retries the operation on a healthy worker. This means that you don't have to worry about your service being unavailable due to a machine failure.
+
+Machines periodically send heartbeats to the control-plane. If a machine fails to send any heartbeats within a certain interval (default 90 seconds):
+
+1. It is marked as unhealthy, and Differential will not send any new requests to it.
+2. The functions in progress are marked as failed, and Differential will retry them on a healthy worker.
+
+If the machine comes back online, Differential will mark it as healthy, and start sending new requests to it. However, it will disregard any results from the machine for the functions that were marked as failed.
```

Comment:
When the machine comes back online

Reply:
I think it's `If`. Sometimes the machine doesn't come online.